### DISCUSSION TITLE
Razzle 4 & After.JS

### DISCUSSION THREAD

**Discussion by anthonyp-usafacts** [jaredpalmer/razzle#1553] [Q&A] [answered]
I've been very happy with using Razzle with [After.js](https://github.com/jaredpalmer/after.js), mostly because it was the same author and the versions were both being supported. However, I haven't seen any updates meaningfully to After.js, nor has the example here been touched in a while. Feels very odd that the only untouched example is the Author's own system.

![image](https://user-images.githubusercontent.com/65727038/107819902-c1aa0380-6d2e-11eb-9f5b-830d91c81825.png)

Can I get confirmation that After.js is being maintained? Or is it a "It just works!" scenario? I'd hate to try to maintain both of these in my platform only to have to gut out After.js once Razzle 4 releases.

Thanks! (A lot!)

**Comment by fivethreeo** [comment]
I am mostly maintaining razzle now. After js is by the author that started razzle. 

Hard to reach him.

Also, take a look at the canary branch, 4.0 razzle is near :)

**Reply by anthonyp-usafacts** [reply]
@fivethreeo I've been following your work and it is VERY appreciated. I've been following the canary tags with a local canary of my server lib, and I'm excited to move onto 4.0 when it releases. My concern here is that Razzle was kinda hand-in-hand with After.js, so this release might make me flush out After.js from my codebase if It's not as maintained.

Seems that jared isn't too available so I should plan safely with After.js.... Thanks!

**Comment by anthonyp-usafacts** [comment]
@fivethreeo I'm seeing a breaking change on Razzle 4 Canary.

Before: https://github.com/jaredpalmer/razzle/blob/master/packages/razzle/config/env.js
```js
        RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest,
        RAZZLE_CHUNKS_MANIFEST: paths.appChunksManifest,
```

After: 
```js
        RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest,
```

After.js relies on this json asset file in order to load the appropriate chunks for the frontend. I'm not even seeing Razzle product the chunks json file at all.

Was there a memo I missed about this change, reason, etc?

* for context, I fixed this easily in my code by using the assets json as both assets and chunks for After.js. Worked fine.

**Comment by fivethreeo** [comment]
Razzle 4 is a major release so we can break stuff ;)

On Thu, 18 Feb 2021 at 23:29, anthonyp-usafacts <notifications@github.com>
wrote:

> @fivethreeo <https://github.com/fivethreeo> I'm seeing a breaking change
> on Razzle 4 Canary.
>
> Before:
> https://github.com/jaredpalmer/razzle/blob/master/packages/razzle/config/env.js
>
>         RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest,
>         RAZZLE_CHUNKS_MANIFEST: paths.appChunksManifest,
>
> After:
>
>         RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest,
>
> After.js relies on this json asset file in order to load the appropriate
> chunks for the frontend. I'm not even seeing Razzle product the chunks json
> file at all.
>
> Was there a memo I missed about this change, reason, etc?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jaredpalmer/razzle/discussions/1553#discussioncomment-382544>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAADGU2I7E5ZM5TELKZXWTS7WIDJANCNFSM4XRJPLQA>
> .
>


-- 
Øyvind Saltvik

**Reply by anthonyp-usafacts** [reply]
> Razzle 4 is a major release so we can break stuff ;)
> […](#)
> On Thu, 18 Feb 2021 at 23:29, anthonyp-usafacts ***@***.***> wrote: @fivethreeo <https://github.com/fivethreeo> I'm seeing a breaking change on Razzle 4 Canary. Before: https://github.com/jaredpalmer/razzle/blob/master/packages/razzle/config/env.js RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest, RAZZLE_CHUNKS_MANIFEST: paths.appChunksManifest, After: RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest, After.js relies on this json asset file in order to load the appropriate chunks for the frontend. I'm not even seeing Razzle product the chunks json file at all. Was there a memo I missed about this change, reason, etc? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#1553 (comment)](https://github.com/jaredpalmer/razzle/discussions/1553#discussioncomment-382544)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAADGU2I7E5ZM5TELKZXWTS7WIDJANCNFSM4XRJPLQA> .
> -- Øyvind Saltvik

Yeah for sure, that's why I opened a Discussion and not an Issue. I get it! I just wanted to know if there was either an existing callout about this breaking change, or (I guess now) if there should be one added to the migration docs.

**Comment by fivethreeo** [comment, marked as answer]
RAZZLE_ASSETS_MANIFEST has that now i think


On Fri, 19 Feb 2021 at 00:31, Fivethreeo <oyvind.saltvik@gmail.com> wrote:

> Razzle 4 is a major release so we can break stuff ;)
>
> On Thu, 18 Feb 2021 at 23:29, anthonyp-usafacts <notifications@github.com>
> wrote:
>
>> @fivethreeo <https://github.com/fivethreeo> I'm seeing a breaking change
>> on Razzle 4 Canary.
>>
>> Before:
>> https://github.com/jaredpalmer/razzle/blob/master/packages/razzle/config/env.js
>>
>>         RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest,
>>         RAZZLE_CHUNKS_MANIFEST: paths.appChunksManifest,
>>
>> After:
>>
>>         RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest,
>>
>> After.js relies on this json asset file in order to load the appropriate
>> chunks for the frontend. I'm not even seeing Razzle product the chunks json
>> file at all.
>>
>> Was there a memo I missed about this change, reason, etc?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/jaredpalmer/razzle/discussions/1553#discussioncomment-382544>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAAADGU2I7E5ZM5TELKZXWTS7WIDJANCNFSM4XRJPLQA>
>> .
>>
>
>
> --
> Øyvind Saltvik
>


-- 
Øyvind Saltvik

**Comment by fivethreeo** [comment]
Yes, should be added.

On Fri, 19 Feb 2021 at 00:45, anthonyp-usafacts <notifications@github.com>
wrote:

> Razzle 4 is a major release so we can break stuff ;)
> … <#m_8628845274860547324_>
> On Thu, 18 Feb 2021 at 23:29, anthonyp-usafacts *@*.***> wrote:
> @fivethreeo <https://github.com/fivethreeo> https://github.com/fivethreeo
> I'm seeing a breaking change on Razzle 4 Canary. Before:
> https://github.com/jaredpalmer/razzle/blob/master/packages/razzle/config/env.js
> RAZZLE_ASSETS_MANIFEST: paths.appAssetsManifest, RAZZLE_CHUNKS_MANIFEST:
> paths.appChunksManifest, After: RAZZLE_ASSETS_MANIFEST:
> paths.appAssetsManifest, After.js relies on this json asset file in order
> to load the appropriate chunks for the frontend. I'm not even seeing Razzle
> product the chunks json file at all. Was there a memo I missed about this
> change, reason, etc? — You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <#1553 (comment)
> <https://github.com/jaredpalmer/razzle/discussions/1553#discussioncomment-382544>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AAAADGU2I7E5ZM5TELKZXWTS7WIDJANCNFSM4XRJPLQA
> .
> -- Øyvind Saltvik
>
> Yeah for sure, that's why I opened a Discussion and not an Issue. I get
> it! I just wanted to know if there was either an existing callout about
> this breaking change, or (I guess now) if there should be one added to the
> migration docs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jaredpalmer/razzle/discussions/1553#discussioncomment-382686>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAADGRWVGGWHKCDECXZIKLS7WRDBANCNFSM4XRJPLQA>
> .
>


-- 
Øyvind Saltvik
